osidb/models.py

```diff
@@ -1952,6 +1952,82 @@ def bzsync(self, *args, bz_api_key=None, **kwargs):
         self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
 
 
+class FlawAcknowledgmentManager(ACLMixinManager, TrackingMixinManager):
+    """flaw acknowledgment manager"""
+
+    @staticmethod
+    def create_flawacknowledgment(flaw, name, affiliation, **extra_fields):
+        """return a new flaw acknowledgment or update an existing flaw acknowledgment without saving"""
+        try:
+            flawacknowledgment = FlawAcknowledgment.objects.get(
+                flaw=flaw,
+                name=name,
+                affiliation=affiliation,
+            )
+            for attr, value in extra_fields.items():
+                setattr(flawacknowledgment, attr, value)
+            return flawacknowledgment
+
+        except ObjectDoesNotExist:
+            return FlawAcknowledgment(
+                flaw=flaw,
+                name=name,
+                affiliation=affiliation,
+                **extra_fields,
+            )
+
+
+class FlawAcknowledgment(ACLMixin, BugzillaSyncMixin, TrackingMixin):
```

Comment:
Could you please describe in docstring how flaw acklowledgments (with public/private flaw source) work?

Reply:
I don't understand. Can you show me an example how that should look, please?

Reply:
Added explanation in docstring.